### PR TITLE
[DNC] Update version to 2023-05-18-preview, migrate to use AAZ

### PR DESCRIPTION
**Pull request by jackieluc**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command

az dnc

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
DNC

**Review by kairu-ms — APPROVED** [review]

**Comment by AllyW** [comment]
> ❌Azure CLI Extensions Breaking Change Test

lost confirmation for aaz cmd. log in https://github.com/Azure/azure-cli-dev-tools/issues/414

File: src/dnc/azext_dnc/aaz/latest/dnc/controller/_wait.py [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+# pylint: skip-file
+# flake8: noqa
+
+from azure.cli.core.aaz import *
+
+
+@register_command(
+    "dnc controller wait",
+)
+class Wait(AAZWaitCommand):
```
Comment (thomasricci) [inline src/dnc/azext_dnc/aaz/latest/dnc/controller/_wait.py:17]: What's the use of this command ?
It seems a GET (pretty similar to the show) with a different name. Does it require some particular logic on the RP?

File: src/dnc/azext_dnc/__init__.py [hunk 2]
```diff
@@ -1,50 +1,42 @@
...
-        parent.__init__(cli_ctx=cli_ctx, custom_command_type=dnc_custom)
+        custom_command_type = CliCommandType(
+            operations_tmpl='azext_dnc.custom#{}')
+        super().__init__(cli_ctx=cli_ctx,
+                         custom_command_type=custom_command_type)
 
     def load_command_table(self, args):
-        from azext_dnc.generated.commands import load_command_table
-        load_command_table(self, args)
+        from azext_dnc.commands import load_command_table
```
[the remark this answers is not in the document]
Comment (jackieluc) [reply src/dnc/azext_dnc/__init__.py:22]: I'm not sure about the details, but this is basically completely replacing the old CLI code with AAZ command models

File: src/dnc/azext_dnc/aaz/latest/dnc/controller/_wait.py (hunk 1 above)
Comment (jackieluc) [reply src/dnc/azext_dnc/aaz/latest/dnc/controller/_wait.py:17]: We don't have any logic in the RP, but this seems to be a global command that is included with AAZ command model. I don't see any way of removing it, but it exists today as well: https://learn.microsoft.com/en-us/cli/azure/dnc/controller?view=azure-cli-latest#az-dnc-controller-wait

**Comment by jackieluc** [comment]
@kairu-ms @jsntcy - Looks like the PR automation identified breaking changes. Are there any blockers to this PR from your perspectives that prevents merging? For some additional context, the `dnc` extension is migrating to use AAZ command models. 

Also, what is the expected timeline for reviewing and merging changes like this?

File: src/dnc/azext_dnc/tests/latest/recordings/test_dnc_controller_create_test_purpose.yaml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+      - application/json
+      ParameterSetName:
+      - --location --resource-name --resource-group --purpose
+      User-Agent:
+      - AZURECLI/2.50.0 (AAZ) azsdk-python-core/1.26.0 Python/3.10.11 (Windows-10-10.0.22621-SP0)
+    method: PUT
+    uri: https://centraluseuap.management.azure.com/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/cli_test_dnc000001/providers/Microsoft.DelegatedNetwork/controller/clitestcontroller?api-version=2023-05-18-preview
+  response:
+    body:
+      string: '{"id":"/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/cli_test_dnc000001/providers/Microsoft.DelegatedNetwork/controller/clitestcontroller","location":"centraluseuap","name":"clitestcontroller","type":"Microsoft.DelegatedNetwork/controller","properties":{"dncAppId":"a91b1853-4403-4f54-b5cb-d1ea19d90c37","dncEndpoint":"https://openai-centraluseuap.dnc.azure.com","controllerProperties":{"dncAppId":"a91b1853-4403-4f54-b5cb-d1ea19d90c37","dncEndpoint":"https://openai-centraluseuap.dnc.azure.com","dncTenantId":null},"resourceGuid":"33b2d4dd-1290-4810-8e16-8283177b4538","provisioningState":"Succeeded"}}'
```
Comment (thomasricci) [inline src/dnc/azext_dnc/tests/latest/recordings/test_dnc_controller_create_test_purpose.yaml:25]: I would add `purpose: "test"` in the response body for consistency

**Review by thomasricci — APPROVED** [review]

**Comment by kairu-ms** [comment]
> @kairu-ms @jsntcy - Looks like the PR automation identified breaking changes. Are there any blockers to this PR from your perspectives that prevents merging? For some additional context, the `dnc` extension is migrating to use AAZ command models.
> 
> Also, what is the expected timeline for reviewing and merging changes like this?

There's no real breaking change in your PR. We will filter out some meaningless messages.

**Comment by jackieluc** [comment]
> > @kairu-ms @jsntcy - Looks like the PR automation identified breaking changes. Are there any blockers to this PR from your perspectives that prevents merging? For some additional context, the `dnc` extension is migrating to use AAZ command models.
> > Also, what is the expected timeline for reviewing and merging changes like this?
> 
> There's no real breaking change in your PR. We will filter out some meaningless messages.

Thanks @kairu-ms, what is the timeline for something like this to merge? Can you clarify what the milestone means? Is that the target date for this to merge?

**Comment by kairu-ms** [comment]
> > > @kairu-ms @jsntcy - Looks like the PR automation identified breaking changes. Are there any blockers to this PR from your perspectives that prevents merging? For some additional context, the `dnc` extension is migrating to use AAZ command models.
> > > Also, what is the expected timeline for reviewing and merging changes like this?
> > 
> > 
> > There's no real breaking change in your PR. We will filter out some meaningless messages.
> 
> Thanks @kairu-ms, what is the timeline for something like this to merge? Can you clarify what the milestone means? Is that the target date for this to merge?

I can merge it anytime. Please fix the CI issues.

File: src/dnc/azext_dnc/tests/latest/recordings/test_dnc_controller_create_test_purpose.yaml (hunk 3 above)
Comment (jackieluc) [reply src/dnc/azext_dnc/tests/latest/recordings/test_dnc_controller_create_test_purpose.yaml:25]: closing because it's no longer needed

**Comment by jackieluc** [comment]
> > > > @kairu-ms @jsntcy - Looks like the PR automation identified breaking changes. Are there any blockers to this PR from your perspectives that prevents merging? For some additional context, the `dnc` extension is migrating to use AAZ command models.
> > > > Also, what is the expected timeline for reviewing and merging changes like this?
> > > 
> > > 
> > > There's no real breaking change in your PR. We will filter out some meaningless messages.
> > 
> > 
> > Thanks @kairu-ms, what is the timeline for something like this to merge? Can you clarify what the milestone means? Is that the target date for this to merge?
> 
> I can merge it anytime. Please fix the CI issues.

Thanks @kairu-ms, the CI is green now. Please merge if there's no other concerns on your end.

**Pull request merged**